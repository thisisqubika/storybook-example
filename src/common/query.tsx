import {
  QueryClient, QueryClientConfig, QueryClientProvider,
} from '@tanstack/react-query';

export function queryClientConfig(
  config?: QueryClientConfig,
): QueryClientConfig {
  return {
    ...config,
    defaultOptions: {
      ...config?.defaultOptions,
      queries: {
        refetchOnWindowFocus: false,
        retry: 1,
        ...config?.defaultOptions?.queries,
      },
    },
  };
} // function newQueryClient

export interface QueryClientWrapperProps {
  config?: QueryClientConfig;
  children: JSX.Element;
} // interface ThemeWrapperProps

export function QueryClientWrapper({
  config,
  children,
}: QueryClientWrapperProps) {
  const queryClient = new QueryClient(queryClientConfig(config));
  return (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );
} // function QueryClientWrapper

