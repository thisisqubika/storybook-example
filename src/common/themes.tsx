// Check out <https://zenoo.github.io/mui-theme-creator/>.
import {
  createTheme, CssBaseline, ThemeProvider, ThemeOptions,
} from '@mui/material';

export const themeOptions: ThemeOptions = {
  palette: {
    mode: 'light',
    primary: {
      main: '#812020',
    },
    secondary: {
      main: '#cc8333',
    },
    background: {
      default: '#e0cab8',
      paper: '#efe8e2',
    },
    error: {
      main: '#fb3636',
    },
    info: {
      main: '#025e90',
    },
    warning: {
      main: '#e49e01',
    },
    success: {
      main: '#2b9430',
    },
  },
  typography: {
    fontFamily: 'Lora',
    fontSize: 13,
  },
}; // const themeOptions

export interface ThemeWrapperProps {
  children: JSX.Element;
} // interface ThemeWrapperProps

export function ThemeWrapper({
  children,
}: ThemeWrapperProps) {
  return (
    <ThemeProvider theme={createTheme(themeOptions)}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  );
} // function ThemeWrapper
