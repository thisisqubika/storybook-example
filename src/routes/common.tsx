import { isRouteErrorResponse, useRouteError } from 'react-router-dom';

export const PATHS = {
  books: 'books',
}; // const PATHS

export function ErrorBoundary() {
  const error: unknown = useRouteError();
  return isRouteErrorResponse(error) ? (
    <h1>{error.status} {error.statusText}</h1>
  ) : (
    <h1>{(error as Error)?.message ?? error}</h1>
  );
} // function ErrorBoundary
