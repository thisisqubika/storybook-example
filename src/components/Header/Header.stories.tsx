import type { Meta, StoryObj } from '@storybook/react';
import { Typography } from '@mui/material';
import { ThemeWrapper } from '../../common/themes';
import { Header } from '.';

export default {
  component: Header,
} satisfies Meta; // meta

export const Empty: StoryObj = {
  render() {
    return (
      <ThemeWrapper>
        <Header />
      </ThemeWrapper>
    );
  },
}; // const Empty

export const Children: StoryObj = {
  render() {
    return (
      <ThemeWrapper>
        <Header>
          <Typography variant='h6'>Sample header text</Typography>
        </Header>
      </ThemeWrapper>
    );
  },
}; // const Empty