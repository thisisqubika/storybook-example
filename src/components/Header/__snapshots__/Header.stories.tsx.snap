// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header Children smoke-test 1`] = `
<div class="MuiBox-root css-i9gxme">
  <header class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-eoy4xr-MuiPaper-root-MuiAppBar-root"
          data-testid="Header"
  >
    <div class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root">
      <h3 class="MuiTypography-root MuiTypography-h3 css-e6ke5q-MuiTypography-root"
          data-testid="Logo"
      >
        Books
      </h3>
      <h6 class="MuiTypography-root MuiTypography-h6 css-rb8u0f-MuiTypography-root">
        Sample header text
      </h6>
    </div>
  </header>
</div>
`;

exports[`components/Header Empty smoke-test 1`] = `
<div class="MuiBox-root css-i9gxme">
  <header class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-eoy4xr-MuiPaper-root-MuiAppBar-root"
          data-testid="Header"
  >
    <div class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root">
      <h3 class="MuiTypography-root MuiTypography-h3 css-e6ke5q-MuiTypography-root"
          data-testid="Logo"
      >
        Books
      </h3>
    </div>
  </header>
</div>
`;
